### ISSUE TITLE
Offline access for tagging sensitive data

### ISSUE THREAD

**Issue by karthim** [heartexlabs/label-studio#4612]
Apologies for my lack of expertise in this matter. I am interested in using Label Studio to annotate sensitive data privately. After installing Label Studio on our local machine, I noticed that it requires an internet connection for authentication and continuously sends data to the labelstud.io server. I'm concerned about the security of our private data and want to ensure that it does not get leaked to the internet during this process.

I would like to know if it is possible to run a fully private instance of Label Studio that operates self-contained without any internet connection. Any insights or guidance on this matter would be greatly appreciated. Thank you!

**Comment by turtle0x1** [comment]
+1, installed recently and its a great tool, but noticed a lot the frontend clicks get tracked (ublock fixed this) and am concerned what the back end then shares.

Is there a flag for disabling sentry.io all together?

**Comment by turtle0x1** [comment]
So it looks like `collect_analytics` has to be set to false in the environment 

https://github.com/HumanSignal/label-studio/blob/15c46d268f24107153ce27cb0f22a3f789ddbf48/label_studio/core/utils/contextlog.py#L191C1-L193

But It looks like if collect_analytics isn't set to false every request send to the backend is then logged of through the core.middleware.ContextLogMiddleware to the remote server?

**Comment by donbobka** [comment]
you also can set strict csp rules + backend firewall policies to avoid external communication

**Comment by turtle0x1** [comment]
Firewall rules are good but I'd also like to have it disabled all together, reduces the false positives pings. The frontend leaks a fair amount of information as well as it isn't sent via the server its sent directly to sentry. Here's the main places I found it can "phone home";

 - Backend ContextLogger (appears to be controllable by env variable `collect_analytics`
 - Backend Sentry (controllable by env, I think)
 - Frontend Sentry (not controllable by env)
 - Frontend google tag manager (not controllable by env)

Did I miss any cases?

Am I correct that setting export `SENTRY_RATE=0`, `SENTRY_DSN=""` prevents the backend sentry logging?

Would a patch be accepted to control the frontend analytics (googletagmanager, sentry) with another env variable? Or perhaps just re-using `collect_analytics` to also disable them in this case?

**Comment by donbobka** [comment]
@turtle0x1 that's why i mentioned CSP. You can set strict CSP (allow only `self` and explicit list of domain) so the browser won't send anything to 3rd party website. Usually it's quite simple to inject csp into http headers: for example in cloudfront if you use aws, or you can inject it directly into nginx of labelstudio docker image
The main problem with disabling it manually one by one using envs or even forking codebase to delete all of them, it doesn't protect from future "loggers". 

 But agree it will be much nicer to have all trackers/loggers to be opt-in

**Comment by AbubakarSaad** [comment]
Hello,

You can set `collect_analytics=0` and that will stop with collecting on everything. Does this help?

**Comment by deniskrumko** [comment]
Hello guys,

So it seems that we can't disable telemetry from Label Studio frontend at all? Env variables like `collect_analytics=0` only work for backend, but each click on webpage spawns like 2 requests to public Sentry from our private network. Each click!

I think that it's not very nice to hide such behaviour from your users in open-source library. I don't think that it's even documented.

Is it possible for you to make telemetry optional? Or we need to fork the project and remove `initSentry` lines by our own? Thanks ✌️

**Comment by nylocx** [comment]
I have the same problem, cant sell this tool internally with analytics functionality included as we have to label pretty sensitive data protected by various NDAs.

**Comment by srajabi** [comment]
Is this still the case that the FE will collect telemetry?

**Comment by makseq** [comment]
No, frontend doesn't collect telemetry. Also if you use collect_analytics=0, nothing is collected.

**Comment by makseq** [comment]
Sentry is used for logging of error exceptions only.
You can try setting `FRONTEND_SENTRY_DSN=`  and `SENTRY_DSN=` to empty strings when you start label studio to disable it on the frontend.

**Comment by csavage5** [comment]
I have the following overriding the Helm chart, and have confirmed the values are being set on the `ls-app` Kubernetes pod:
```
global: 
  extraEnvironmentVars:
    collect_analytics: "False" 
    SENTRY_RATE: "0"
    SENTRY_DSN: ""
    FRONTEND_SENTRY_DSN: ""
```

With these values set, external traffic is still observed to an IP address that points to an AWS instance. I've also tried setting `collect_analytics` to `"0"` with the same result.

If I pass `" "` or `"None"` to `SENTRY_DSN` and `FRONTEND_SENTRY_DSN` instead, the init container `db_migrations` crashes from the Sentry SDK library with:

`sentry_sdk.utils.BadDsn: Unsupported scheme ''`

**Comment by csavage5** [comment]
To follow-up, I'm seeing traffic to the following IPs with the above settings, I'm assuming this is tele.labelstud.io as the IPs resolve to an AWS cluster.

<img width="388" alt="Screenshot 2024-04-22 at 11 10 08 AM" src="https://github.com/HumanSignal/label-studio/assets/22351542/5f862c34-a5f1-40b3-bd96-5d8739b23801">

**Comment by charlesverge** [comment]
For sentry there appears to be no configuration option to disable it or use a different sentry configuration.
https://github.com/HumanSignal/label-studio/blob/066f74b2872b69b5285fc7ae735ed9192865ba1b/web/apps/labelstudio/src/app/App.jsx#L38

https://github.com/HumanSignal/label-studio/blob/066f74b2872b69b5285fc7ae735ed9192865ba1b/web/apps/labelstudio/src/config/Sentry.ts#L7


To disable Sentry a potential solution is to create a new setting in APP_SETTINGS or an environment variable to disable this feature.

Note in this configuration data like users email and username are passed to label studio which may not be desired for everyone

tele.labelstud.io is referenced in https://github.com/HumanSignal/label-studio/blob/066f74b2872b69b5285fc7ae735ed9192865ba1b/label_studio/core/utils/contextlog.py#L311

Which is loaded as part of the middle ware in https://github.com/HumanSignal/label-studio/blob/066f74b2872b69b5285fc7ae735ed9192865ba1b/label_studio/core/settings/base.py#L242

In theory if you are in dire need to disable this tracking you could comment out much of this code at it appears if the application would work fine.

**Comment by csavage5** [comment]
I've figured out why setting `collect_analytics` in Helm wasn't working, and raised the above bug report. I've raised a PR to fix: https://github.com/HumanSignal/label-studio/pull/5788

**Comment by makseq** [comment]
The PR is merged. It will be available starting form LS 1.13. Right now you can use the develop branch.
